src/components/Autosuggest/Autosuggest.js

```diff
@@ -14,23 +14,27 @@ import {
     ENTER_KEY,
 } from '../../constants';
 
+const FOCUS_DELAY = 250;
+
 const { block, elem } = bem('Autosuggest', styles);
 
 class Autosuggest extends React.Component {
     constructor(props) {
         super(props);
-        this.inputRef = React.createRef();
-        this.rootRef = React.createRef();
-        this.listRef = React.createRef();
+
+        this.inputRef = props.inputRef || React.createRef();
+        this.rootRef = props.rootRef || React.createRef();
+        this.listRef = props.listRef || React.createRef();
+
+        this.handleTagDeleteClick = memoize(this.handleTagDeleteClick);
+        this.handleWrapperClick = memoize(this.handleWrapperClick);
+        this.handleWrapperKeyDown = memoize(this.handleWrapperKeyDown);
```

Comment:
Seems memoization doesn't work correctly here. I typed the same in input several times but function `this.handleWrapperKeyDown` was invoked on every change.

Reply:
@mukiienko I didn't quite understand why it should not be invoked on each change, could you clarify it please? And also it seems to be a class component which we should split onto a few separate ones with rewriting it to hooks logic, so memoization will go away soon anyway.

Reply:
I thought it needs for memoization result, but it memoizes reference on the function. I was mistaken :)